### PR TITLE
Add `target_format` property to the Logs attribute remapper 

### PR DESCRIPTION
**Pull request by vinckama**
Add target_format to attribute remapper
related to [PR](https://github.com/DataDog/datadog-api-spec/pull/587)

### CODE REVIEW

File: docs/resources/logs_custom_pipeline.md [hunk 1]
```diff
@@ -185,6 +186,7 @@ The following arguments are supported:
   - `source_type`: (Required) Defines where the sources are from (log `attribute` or `tag`).
   - `target`: (Required) Final `attribute` or `tag` name to remap the sources.
   - `target_type`: (Required) Defines if the target is a log `attribute` or `tag`.
+  - `target_format`: (Optional) If the target of the remapper is an attribute, try to cast the value to a new specific type. If the cast is not possible, the original type is kept. `string`, `integer`, or `double` are the possible types.
```
Comment (txominpelu) [inline docs/resources/logs_custom_pipeline.md:189]: Maybe you can make it more explicit that users should only set the target_format if the target_type is an attribute. It's already said, but for this kind of valid / invalid values it's better to be super explicit and not being afraid of being repetitive.

**Review by txominpelu — APPROVED** [review]

**Review by txominpelu — APPROVED** [review]

File: docs/resources/logs_custom_pipeline.md [hunk 2]
```diff
@@ -185,12 +186,13 @@ The following arguments are supported:
...
   - `target`: (Required) Final `attribute` or `tag` name to remap the sources.
   - `target_type`: (Required) Defines if the target is a log `attribute` or `tag`.
+  - `target_format`: (Optional) If the `target_type` of the remapper is an attribute, try to cast the value to a new specific type. If the cast is not possible, the original type is kept. `string`, `integer`, or `double` are the possible types. If the `target_type is a `tag`, it should not be specified, otherwise it will raise an issue.
   - `preserve_source`: (Optional, default = false) Remove or preserve the remapped source element.
   - `override_on_conflict`: (Optional, default = false) Override the target element if already set.
   - `name`: (Optional) Name of the processor
   - `is_enabled`: (Optional, default = false) If the processor is enabled or not.
 - category_processor
-  - `target`: (Required) Name of the target attribute whose value is defined by the matching category.
+  - `target`: (Required) Name of the target attribute whose value is defined by the matching category.`
```
Comment (therve) [inline docs/resources/logs_custom_pipeline.md:195]: Looks like an unintended change?

File: docs/resources/logs_custom_pipeline.md [hunk 3]
```diff
@@ -31,7 +31,8 @@ resource "datadog_logs_custom_pipeline" "sample_pipeline" {
             sources = ["db.instance"]
             source_type = "tag"
             target = "db"
-            target_type = "tag"
+            target_type = "attribute"
+            target_format = "String"
```
Comment (therve) [inline docs/resources/logs_custom_pipeline.md:35]:
```suggestion
            target_format = "string"
```

**Review by therve — CHANGES_REQUESTED** [review]
Looks good, some minor doc changes. Thanks!

File: docs/resources/logs_custom_pipeline.md (hunk 2 above)
Comment (vinckama) [reply docs/resources/logs_custom_pipeline.md]: yes thanks 👌

File: docs/resources/logs_custom_pipeline.md (hunk 1 above)
Comment (vinckama) [reply docs/resources/logs_custom_pipeline.md]: thank you for pointing this out

**Review by therve — APPROVED** [review]

**Comment by gmetzker** [comment]
Any idea when this might be released?  This feature is exactly what I was hunting for in the docs.

**Comment by therve** [comment]
Hello, we'll release in the next few weeks.

**Pull request merged**